src/cosmos_provider.js

```diff
@@ -0,0 +1,141 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+"use strict";
+
+import BaseProvider from "./base_provider";
+import Utils from "./utils";
+import ProviderRpcError from "./error";
+import { Buffer } from "buffer";
+const { CosmJSOfflineSigner, CosmJSOfflineSignerOnlyAmino } = require('./cosmjs');
+
+export class TrustCosmosWeb3Provider extends BaseProvider {
+  constructor(config) {
+    super(config);
+
+    this.providerNetwork = "cosmos";
+    this.callbacks = new Map();
+    this.mode = "extension";
+    this.isKeplr = true;
+    this.version = "0.10.16";
+  }
+
+  enable(chainIds) {
```

Comment:
_The window.keplr.enable(chainIds) method requests the extension to be unlocked if it's currently locked. If the user hasn't given permission to the webpage, it will ask the user to give permission for the webpage to access Keplr._

I don't think it applies to mobile wallets much, so I skipped it. We handle this permission logic via `requestAccounts`.

Reply:
no callback needed?

Reply:
nope https://github.com/chainapsis/keplr-wallet/blob/0977296c7d489cbde8bf372350c95a77dc0fd036/packages/provider/src/inject.ts#L211